src/main/java/chatbot/parser/Parser.java

```diff
@@ -0,0 +1,84 @@
+package chatbot.parser;
+
+import chatbot.commands.Command;
+import chatbot.commands.EventCommand;
+import chatbot.commands.ExitCommand;
+import chatbot.commands.TodoCommand;
+import chatbot.commands.DeadlineCommand;
+import chatbot.commands.ListCommand;
+import chatbot.commands.DeleteCommand;
+import chatbot.commands.MarkCommand;
+import chatbot.commands.UnmarkCommand;
+import chatbot.commands.FindCommand;
+import chatbot.exceptions.DukeException;
+
+/**
+ * Represents the class that is responsible for parsing user inputs and generating the relevant commands
+ */
+public class Parser {
+    /**
+     * Checks if the user input is valid
+     * @param command User input to be validated
+     * @return Returns a boolean that indicates if the input is valid
+     * @throws DukeException If the user input is not valid
+     */
+    public static boolean isValid (String command) throws DukeException {
+        if (command.equals("list") || command.equals("bye")) {
+            return true;
+        } else if (command.split(" ")[0].equals("todo") && command.split(" ")[1].length() < 1) {
+            throw new DukeException("Error! Please include a description of the todo");
+        } else if (command.split(" ")[0].equals("deadline") && command.split(" ")[1].length() < 1
+                && !command.contains("/by")) {
+            throw new DukeException("Error! Input should have a description and date");
+        } else if (command.split(" ")[0].equals("event") && command.split(" ")[1].length() < 1
+                && !command.contains("/from") && !command.contains("/to")) {
+            throw new DukeException("Error! Input should have a description, start date and end date");
+        } else if (command.split(" ")[0].equals("mark") && command.split(" ")[1].length() < 1) {
+            throw new DukeException("Error! Please include a description of the task to be marked");
+        } else if (command.split(" ")[0].equals("unmark") && command.split(" ")[1].length() < 1) {
+            throw new DukeException("Error! Please include a description of the task to be unmarked");
+        } else {
```

Comment:
Just nitpicking a little but you can consider adding the format that the user is supposed to enter so that they have a point of reference, since most of the error messages are generic for the particular task entered and the user may not know what went wrong.